core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts

```diff
@@ -122,6 +122,29 @@ export class DotPageStateService {
      * @memberof DotPageStateService
      */
     setInternalNavigationState(state: DotPageRenderState): void {
+        const urlParam = generateDotFavoritePageUrl({
+            deviceInode: state.viewAs.device?.inode,
+            languageId: state.viewAs.language.id,
+            pageURI: state.page.pageURI,
+            siteId: state.site?.identifier
+        });
+
+        this.dotESContentService
+            .get({
+                itemsPerPage: 10,
+                offset: '0',
+                query: `+contentType:DotFavoritePage +deleted:false +working:true +owner:${state.user.userId} +DotFavoritePage.url_dotraw:${urlParam}`
+            })
+            .pipe(take(1))
+            .subscribe((response: ESContent) => {
+                const favoritePage = response.jsonObjectView?.contentlets[0];
+
+                if (favoritePage) {
+                    state.favoritePage = favoritePage;
+                    this.setCurrentState(state);
+                }
+            });
```

Comment:
Should this have an error handler? @fmontes

Reply:
Yes... and we can't interrupt the user with the error, if we can't get the fav, we log the error in the console and let the user continue its path.

Reply:
Don't we need a catch here? if the requets fail the app crash, right?

Reply:
not needed, previously the local internal state is set, so if that favorite request fails, the part of the state of favorites is the only that won't be updated, so it won't break the navigation/Edit mode